projects/gateway2/query/httproute.go

```diff
@@ -408,6 +421,24 @@ func (r *gatewayQueries) GetRoutesForGateway(ctx context.Context, gw *gwv1.Gatew
 	return ret, nil
 }
 
+func (r *gatewayQueries) wildcardNamespaceExists(ctx context.Context) (bool, error) {
+	ns := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: wellknown.RouteDelegationLabelSelectorWildcardNamespace,
+		},
+	}
+	err := r.client.Get(ctx, types.NamespacedName{Name: ns.Name}, ns)
+	if err == nil {
+		// Namespace exists
+		return true, nil
+	} else if err != nil && k8serrors.IsNotFound(err) {
```

Comment:
nit: `err != nil` will always be true in this case so you don't have to check it

Reply:
good catch, missed this when I refactored the code